### ISSUE TITLE
Large sprites are not displayed correctly.

### ISSUE THREAD

**Issue by Djeeshka** [CleverRaven/Cataclysm-DDA#66280]
### Describe the bug

Part of a large sprite that occupy a right in-game cell from their origin does not show up:
![image](https://github.com/CleverRaven/Cataclysm-DDA/assets/24527469/f45c4c3d-b6b9-4d54-825a-13405c5c4412)
![image](https://github.com/CleverRaven/Cataclysm-DDA/assets/24527469/4b6ebeb9-56f3-45ca-8989-15ae63296583)
Tileset doesn't matter. Renderer doesn't matter. Scaling factor, scaling mode, batching does not matter as well.
Even when sprite is flipped:
![image](https://github.com/CleverRaven/Cataclysm-DDA/assets/24527469/617e4001-bfef-44db-ace8-c25a61a3e85b)

_Noticed same behavior on screenshots from other people in the discord channel._
Harder to notice with non monster sprites, but if you'll look closely can be noticed on trees like pines and such.

### Attach save file

No special save required.

### Steps to reproduce

1. Have tileset with large sprites like MSXotto+ or UltiCa selected.
2. Spawn a large creature like frogmother, moose, skeletal juggernaut, etc.
3. Observe.

### Expected behavior

Should render that corner tile ¯\_(ツ)_/¯

### Versions and configuration

- OS: Windows 10
- Game Version: b0bf856
- Graphics version: Tiles
- Ingame language: English
- Mods loaded: Dark Days Ahead [dda]

**Comment by andrei8l** [comment]
/confirmed It was indeed caused by #66055

**Comment by Qrox** [comment]
@Rewryte could you look into this?

**Comment by Rewryte** [comment]
Looking into it.  I believe the root cause goes back to #65738, but it is mitigated by the "tile per layer" draw order from the same PR.  When #66055 reverted drawing back to "layer per tile", the bug becomes apparent.

**Comment by Qrox** [comment]
I believe the drawing order before #65738 was `for each row { for each layer { for each tile } }`. With z-levels added, it should probably be `for each z-level { for each row { for each layer { for each tile } } }`, if we assume the sprites may only extend beyond the left, top, and right tile borders. (In other words, lower z-levels are always behind higher z-levels, farther (top) rows are always behind nearer (bottom) rows in a z-level, and lower layers are always behind upper layers in a row.)

The iso mode drawing order was also changed so it may have simiar problems.

**Comment by Rewryte** [comment]
You're right.  It was "tile per layer".  It looks like contrary to my previous notion, #66052 was not caused by a shift away from a "layer per tile" draw order, but the clumping of all rows together when I catalogued them by z-level into `draw_points_3d`.  I might have some ideas on how to improve on this.
